src/dbt_score/model_filter.py

```diff
@@ -0,0 +1,115 @@
+"""Model filtering to choose when to apply specific rules."""
+
+from typing import Any, Callable, Type, TypeAlias, overload
+
+from dbt_score.models import Model
+
+FilterEvaluationType: TypeAlias = Callable[[Model], bool]
+
+
+class ModelFilter:
+    """The Filter base class."""
+
+    description: str
+
+    def __init__(self) -> None:
+        """Initialize the filter."""
+        pass
+
+    def __init_subclass__(cls, **kwargs) -> None:  # type: ignore
+        """Initializes the subclass."""
+        super().__init_subclass__(**kwargs)
+        if not hasattr(cls, "description"):
+            raise AttributeError("Subclass must define class attribute `description`.")
+
+    def evaluate(self, model: Model) -> bool:
```

Comment:
I think it would make sense to rename `evaluate` for `ModelFilter`. Maybe just `filter`?

Reply:
I don't really like `evaluate` either, just kept consistent with `Rule`. Not opposed to changing it at all. However `filter` is not a great name either: `filter(Callable[[...], bool], Iterable[T]) -> Iterable[T]`, and `evaluate(...) -> bool` is just the first parameter of filter, not the higher-order filter function.

That is to say, the name filter for this function conflicts with the semantics of Python's built-in `filter`.

Reply:
Right `filter` is not a good idea either! :sweat_smile: Maybe @matthieucan has a good idea?

Reply:
No strong opinion, but can suggest `apply` for consideration.

Reply:
I think `evaluate` is pretty good, as it's descriptive of what's happening. This function answers a yes/no question about a given model, which is fundamentally an evaluation?
`apply` works as well!